Sources/TwitchIRC/IncomingMessage/PrivateMessage.swift

```diff
@@ -1,6 +1,31 @@
 
 /// A Twitch `PRIVMSG` message.
 public struct PrivateMessage: MessageWithBadges {
+
+    public struct EmoteReference: Identifiable, Equatable {
```

Comment:
Rename to `Emote`.
Remove the `Identifiable`/`Equatable` conformances. It's not this library's responsibility to provide these conformances. You can extensions for these conformances in your own apps.
Move `Emote` struct to its own file in `IncomingMessage` directory, can be called `Emote.swift`.
Then use this for parsing emotes for `var emotes` in 2 other files: `UserNotice.swift` and `Whisper.swift`.